dev/bots/run_command.dart

```diff
@@ -133,9 +142,12 @@ Future<void> runCommand(String executable, List<String> arguments, {
     if (failureMessage != null) {
       print(failureMessage);
     }
-    if (!printOutput) {
-      stdout.writeln(utf8.decode((await savedStdout).expand<int>((List<int> ints) => ints).toList()));
-      stderr.writeln(utf8.decode((await savedStderr).expand<int>((List<int> ints) => ints).toList()));
+
+    // Print the output when we get unexpected results (unless output was
+    // printed already).
+    if (outputMode != OutputMode.print) {
+      stdout.writeln(flattenToString(await savedStdout));
```

Comment:
This change seems unrelated

Reply:
Which part of it?  The `flattenToString` business is there because I wanted to reuse the same transformation that this code path was using (lines 133-134), and I didn't want to duplicate it.